FlagsmithClient/src/main/java/com/flagsmith/api/GetFlags.kt

```diff
@@ -0,0 +1,53 @@
+package com.flagsmith.api
+
+
+import android.net.Uri
+import com.flagsmith.builder.Flagsmith
+import com.flagsmith.response.Flag
+import com.flagsmith.interfaces.INetworkListener
+import com.flagsmith.android.network.NetworkFlag
+import com.flagsmith.android.network.ApiManager
+import com.flagsmith.interfaces.IFlagArrayResult
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import java.net.URLEncoder
+
+class GetFlags(builder: Flagsmith, finish: IFlagArrayResult) {
```

Comment:
I'm not entirely sure why each API request needs it's own class? Shouldn't it just be a single class that manages all requests? If we want to manage it this way, with a separate class per endpoint request, shouldn't there be an interface that each of the classes implement?

Reply:
This is now more in-line with the iOS SDK through the use of a different HTTP client